### ISSUE TITLE
Issue Detecting PNG

### ISSUE THREAD

**Issue by patrickheeney** [sindresorhus/file-type#442]
I have a third party vendor that is providing images that I am attempting to detect what format. The PNG opens fine on a mac, however I get an error when using the library:

```
EndOfStreamError: End-Of-Stream
          at BufferTokenizer.peekToken (/node_modules/strtok3/lib/BufferTokenizer.js:92:19)
          at BufferTokenizer.readToken (/node_modules/strtok3/lib/BufferTokenizer.js:81:29)
          at readChunkHeader (/file-type/core.js:964:27)
          at _fromTokenizer (/file-type/core.js:970:18)
```

When I open the PNG and re-save, it appears to work correctly. I am unaware of how the original images are being generated in this way but they appear to be valid even before re-saving.

```shell
wget -O image https://images2.imgbox.com/98/88/UbYCwMuO_o.png?download=true
```

```js
const FileType = require('file-type');
const readChunk = require('read-chunk');

(async () => {
	const buffer = readChunk.sync('image', 0, 4100);
	console.log(await FileType.fromBuffer(buffer));
	//=> {ext: 'png', mime: 'image/png'}
})();
```

It seems that the ```tokenizer.position``` is 2-4 bytes from the length on the last iteration of ```while (tokenizer.position < tokenizer.fileInfo.size)``` which causes it to attempt to read past the length. I can't quite figure out where it is ending up 2-4 bytes short so the final iteration of the do while loop doesn't happen.

EDIT. Interestingly when I change the chunk length to ```const buffer = readChunk.sync('./image', 0, 4097);``` it works, anything above that it fails. I got that line from the README.

**Comment by Borewit** [comment]
I am sorry, you need to describe a way to reproduce the issue. Please double check you are using streams correctly, you can only read from a stream once.

**Comment by patrickheeney** [comment]
@Borewit Did you see the steps I listed in the original issue for reproducing the issue? You run the `wget` command and the script I posted. Do you not see the error? What output do you get?

**Comment by patrickheeney** [comment]
@Borewit Here is the full step by step reproduction:

```
mkdir ~/Dev/test
cd ~/Dev/test
npm install file-type@15.0.0 read-chunk@3.2.0
wget -O image https://images2.imgbox.com/98/88/UbYCwMuO_o.png?download=true
echo "const FileType = require('file-type');
const readChunk = require('read-chunk');

(async () => {
	const buffer = readChunk.sync('image', 0, 4100);
	console.log(await FileType.fromBuffer(buffer));
	//=> {ext: 'png', mime: 'image/png'}
})();" >> test.js
node test.js
```

EDIT. When I try the latest version it might be OK. Let me know if you see the problem on 15.0.0?

**Comment by patrickheeney** [comment]
It was fixed in ec88386d940f0b9024962465a56cd6e054b5a328 . I had some cached decencies that were using version 15.0.0. Sorry about that and thanks for the fix last year.

**Comment by Borewit** [comment]
Glad that everything is working. 

Duplicate of issue #391.
